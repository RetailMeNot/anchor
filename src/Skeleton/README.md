# Skeleton

The Skeleton component is intended to show placeholder shapes
while a component loads so that the user knows about what to expect
and understands that the content is loading.

If the Skeleton is set to `loading={false}` it just renders it's children. If `loading={true}`, it hides all of it's children.

## Usage

There are approximately 3 ways to use this component:

### Text

If the Skeleton component's child is a string (or textLength is defined) then it displays long gray bars in place of the text. This is to simulate a line of impending text or a paragraph. The longer the child text (or textLength), the longer the skeleton text.

Skeleton text observes font styles (font-size, line-height, etc) because it is implemented via text characters. This means it's a good idea to give it the same styles that you would give to the text it's meant to simulate.

In practice that means you'll probably want to use it with the Skeleton inside your typographical element like so:

```jsx
<Typography tag="h2">
    <Skeleton>I'm the text the skeleton will cover</Skeleton>
</Typography>
```

Text Skeletons inherit their `display` property because in some situations it is needed to format the text properly (e.g. h1s are `block`s, and spans are `inline` and this makes a difference to text's line-height). If the parent isn't a typographical element, you may need to define the `display` prop manually to get it to look just the way you want.

### Wrapping

The second way to use the Skeleton is by wrapping an element that you want to cover. It's as simple as that. For instance, if you had a fixed size img that was still loading, you could wrap it with a Skeleton and it'd simply take the shape of your image and hide it.

```jsx
<Skeleton>
    <Image width={200} height={120} />
</Skeleton>
```

If the wrapped child has a border-radius, you'll probably want to give the the skeleton the same border-radius (especially if the child is circular, like an avatar).

The Skeleton does not know the display setting of the child it's wrapping, so you may need to set it manually (i.e. if the child is a `block` you'll probably want to set the Skeleton to a block as well, so that it fits in the DOM the same way).

### Custom Shape

You might just want to give the Skeleton specific dimensions to make it take the exact shape you desire.

```jsx
<Skeleton width="100%" height="0.5rem" borderRadius="circular" marginBottom="1rem" />
<Skeleton width="100%" height="0.5rem" borderRadius="circular" marginBottom="1rem" />
<Skeleton width="80%" height="0.5rem" borderRadius="circular" />
```
