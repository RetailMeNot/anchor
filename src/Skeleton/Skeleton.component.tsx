// REACT
import * as React from 'react';
// VENDOR
import classNames from 'classnames';
import styled, { css } from '@xstyled/styled-components';
import { th, space as spaceStyles, SpaceProps } from '@xstyled/system';

interface SkeletonProps
    extends React.HTMLAttributes<HTMLDivElement>,
        SpaceProps {
    className?: string;
    loading?: boolean;
    borderRadius?: string;
    display?: string;
    height?: string;
    width?: string;
    maxHeight?: string;
    maxWidth?: string;
    minHeight?: string;
    minWidth?: string;
    textLength?: number;
}

interface StyledSkeletonProps extends SkeletonProps {
    textOnly?: boolean;
}

const StyledSkeleton = styled('div')<StyledSkeletonProps>`
    box-sizing: border-box;

    ${({
        display,
        borderRadius,
        height,
        width,
        maxHeight,
        maxWidth,
        minHeight,
        minWidth,
        textOnly,
    }) =>
        css({
            borderRadius,
            display: display || (textOnly ? 'inherit' : 'inline-block'),
            width,
            height,
            maxWidth,
            maxHeight,
            minHeight,
            minWidth,
        })}
    ${spaceStyles}

    ${({ textOnly }) =>
        textOnly
            ? css`
                  word-break: break-all;

                  color: neutrals.silver.dark;
                  animation: color-change 2s ease-in-out infinite;
              `
            : css`
                  pointer-events: none;
                  background: ${th.color('neutrals.silver.dark')};
                  animation: background-change 2s ease-in-out infinite;

                  // hide all children
                  && * {
                      visibility: hidden;
                      opacity: 0;
                  }
              `}

    @keyframes background-change {
        0% {
            background: ${th.color('neutrals.silver.dark')};
        }
        50% {
            background: ${th.color('neutrals.ash.light')};
        }
        100% {
            background: ${th.color('neutrals.silver.dark')};
        }
    }
    @keyframes color-change {
        0% {
            color: neutrals.silver.dark;
        }
        50% {
            color: neutrals.ash.light;
        }
        100% {
            color: neutrals.silver.dark;
        }
    }
`;

export const Skeleton = ({
    className,
    children,
    textLength,
    loading = true,
    ...props
}: SkeletonProps): React.ReactElement<StyledSkeletonProps> | any => {
    if (!loading) {
        return typeof children === 'undefined' ? null : children;
    }

    // This is the conversion from block width to the average width of
    // a normal distribution of letters. We came up with it by eyeballing it
    // for the Avenir Next font. We should make it configurable via theme/prop
    // or make further calculations to improve it.
    const blockToAverageCharQuotient = 2.15;

    const onlyChild =
        React.Children.count(children) === 1 &&
        React.Children.toArray(children)[0];

    const textOnly = !!textLength || typeof onlyChild === 'string';
    const length =
        textLength || (typeof onlyChild === 'string' && onlyChild.length);
    const blockText =
        length &&
        Array(Math.ceil(length / blockToAverageCharQuotient) + 1).join('â–†');

    return (
        <StyledSkeleton
            className={classNames('anchor-skeleton', className)}
            textOnly={textOnly}
            {...props}
        >
            {textOnly ? blockText : children}
        </StyledSkeleton>
    );
};
