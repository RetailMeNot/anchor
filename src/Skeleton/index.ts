export { Skeleton } from './Skeleton.component';
