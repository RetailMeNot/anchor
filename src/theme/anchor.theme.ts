// ANCHOR THEME
import { radii } from './sizes.theme';
import { MODAL_THEME_KEY, MODAL_THEME } from '../Modal';

// Copied from Adapter.StandardBreakpoints mins
// todo: dedup- they should use a single source
const breakpoints = {
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
};

interface AnchorTheme {
    colors?: any;
    breakpoints?: any;
    radii?: any;
    [key: string]: any;
}

export const anchorTheme: AnchorTheme = {
    breakpoints,
    radii,
    [MODAL_THEME_KEY]: MODAL_THEME,
};
