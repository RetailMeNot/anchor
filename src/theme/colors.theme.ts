export default {
  neutrals: {
    charcoal: {
      base: '#323232',
      light: '#595959',
      dark: '#000000',
    },
    ash: {
      base: '#A6A6A6',
      light: '#D3D3D3',
      dark: '#808080',
    },
    silver: {
      base: '#F1F1F1',
      light: '#FAFAFA',
      dark: '#E7E7E7',
    },
    lavenderWhispers: {
      base: '#F7F5FC',
      light: '#FBFAFF',
      dark: '#F1EEFA',
    },
    white: {
      base: '#FFFFFF',
    },
  },
  main: {
    grapePurchase: {
      base: '#602D6C',
      light: '#91609C',
      dark: '#4B2354',
    },
    rx: {
      base: '#F23466',
      light: '#FF6B9C',
      dark: '#CC2D52',
    },
    genie: {
      base: '#4C18F3',
      light: '#7855D9',
      dark: '#42199A',
    },
  },
  accent: {
    savvyCyan: {
      base: '#0998D6',
      light: '#66CCFF',
      dark: '#0074A6',
    },
    tealBreaker: {
      base: '#00CCCD',
      light: '#00E0D1',
      dark: '#00BDBD',
    },
  },
  secondary: {
    dealEnvy: {
      base: '#14A382',
      light: '#00CDAD',
      dark: '#268068',
    },
    flashPink: {
      base: '#DF206C',
      light: '#E665A1',
      dark: '#B81D5B',
    },
    cyberMango: {
      base: '#EB663D',
      light: '#FA9F5A',
      dark: '#C74E30',
    },
    goldMine: {
      base: '#F5A623',
      light: '#FFC761',
      dark: '#CC861D',
    },
    fireSale: {
      base: '#EA4545',
      light: '#F16667',
      dark: '#CC3B3B',
    },
  },
  alert: {
    error: {
      base: '#CC3B3B',
    },
    success: {
      base: '#268068',
    },
  },
  rx: {
    purple: {
      base: '#A678B5',
    },
  },
};
