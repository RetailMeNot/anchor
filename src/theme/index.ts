export { default as colors } from './colors.theme';
export { default as fonts } from './fonts.theme';
export { default as sizes } from './sizes.theme';
export { default as variables } from './variables.theme';

export { NormalizeCSS, GlobalCSS } from './GlobalStyles';
