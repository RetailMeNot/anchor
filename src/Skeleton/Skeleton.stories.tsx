// REACT
import * as React from 'react';
// STORYBOOK
import { storiesOf } from '@storybook/react';
import { boolean, select, text } from '@storybook/addon-knobs';
// VENDOR
import styled, { ThemeProvider } from '@xstyled/styled-components';
import { th } from '@xstyled/system';
// ANCHOR
import * as README from './README.md';
import { Typography, Avatar, Button, Close } from '..';
import { RootTheme } from '../theme';
import { typography, TypographyTags, Scale } from '../theme/typography.theme';
import { Skeleton } from './Skeleton.component';

const StyledStory = styled('div')`
    padding: 2rem 4rem;
    color: text.body;
    font-family: ${th('typography.fontFamily')};
`;

const Card = styled('div')`
    width: 330px;
    background-color: white;
    border-radius: base;
    box-shadow: 0px 1px 4px 0px rgba(0, 0, 0, 0.2);
    display: flex;
    flex-direction: column;
`;

const CardHeader = styled('div')`
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 1rem;
    position: relative;

    button {
        position: absolute;
        top: 0.5rem;
        right: 0.5rem;
    }

    .anchor-avatar {
        background-color: #ec5673;
    }
`;

const CardInfo = styled('div')`
    margin-left: 1rem;
    display: flex;
    flex-grow: 1;
    flex-direction: column;
    justify-content: center;
`;

const CardImage = styled('div')`
    width: 100%;
    height: 190px;
    background: linear-gradient(170deg, #df206c 0%, #f16667 100%);
`;
const CardContent = styled('div')`
    width: 100%;
`;
const CardFooter = styled('div')`
    padding: 1rem 1rem 1.5rem;
`;

type TagSelect = TypographyTags | '';
type ScaleSelect = Scale | '';
const Tags = [''].concat(Object.keys(typography.tag)) as TagSelect[];
const Scales = ['']
    .concat(Object.keys(typography.scale))
    .map(x => parseInt(x, 10) || '') as ScaleSelect[];

storiesOf('Components/Skeleton', module)
    .addParameters({
        readme: {
            sidebar: README,
        },
    })
    .add('Text', () => {
        const loading = boolean('loading', true);
        const defaultText =
            // tslint:disable-next-line: max-line-length
            'For a long time in software companies the practice of building software as one-off initiatives at has been called into question and deemed insular. Its no longer acceptable to deliver a piece of software that solves a particular problem and call it a solution. As new requirements emerge, narrowing our focus just to a particular challenges requirements results in organizational tension, substandard solutions, churn, code rot, technical debt, etc.';

        const textLength = parseInt(text('textLength', ''), 10) || undefined;
        const tag = select<TagSelect>('Typography tag', Tags, '');
        const scale = select<ScaleSelect>('Typography scale', Scales, '');

        return (
            <ThemeProvider theme={RootTheme}>
                <StyledStory>
                    <Typography
                        tag={tag || undefined}
                        scale={scale || undefined}
                    >
                        <Skeleton loading={loading} textLength={textLength}>
                            {text('Text', defaultText)}
                        </Skeleton>
                    </Typography>
                </StyledStory>
            </ThemeProvider>
        );
    })
    .add('Custom Shape', () => {
        const loading = boolean('loading', true);

        return (
            <ThemeProvider theme={RootTheme}>
                <StyledStory>
                    <Skeleton
                        loading={loading}
                        display={text('display', '') || undefined}
                        borderRadius={text('borderRadius', 'circular')}
                        width={text('width', '10rem') || undefined}
                        height={text('height', '2rem') || undefined}
                        maxWidth={text('maxWidth', '') || undefined}
                        maxHeight={text('maxHeight', '') || undefined}
                        minWidth={text('minWidth', '') || undefined}
                        minHeight={text('minHeight', '') || undefined}
                        margin={text('margin', '') || undefined}
                        padding={text('padding', '') || undefined}
                        textLength={
                            parseInt(text('textLength', ''), 10) || undefined
                        }
                    />
                </StyledStory>
            </ThemeProvider>
        );
    })
    .add('Default', () => {
        const loading = boolean('loading', true);

        const fetchedData = {
            title: text('title', 'Keanu Reeves'),
            subtitle: text('subtitle', 'Actor, Musician'),
            description: text(
                'description',
                'Keanu Charles Reeves is a Canadian actor and musician. You may know him by his real name, John Wick.'
            ),
            initials: text('initials', 'KR'),
        };
        const placeholderData = {
            title: text('loading.title', 'Friendly title'),
            subtitle: text('loading.subtitle', 'Lovely subtitle'),
            description: text(
                'loading.description',
                "The text here doesn't actually matter, its just used to determine how long to make the skeleton text."
            ),
            initials: text('loading.initials', '') || undefined,
        };

        const { title, subtitle, initials, description } = loading
            ? placeholderData
            : fetchedData;

        return (
            <ThemeProvider theme={RootTheme}>
                <StyledStory>
                    <Card>
                        <CardHeader>
                            <Skeleton loading={loading} borderRadius="circular">
                                <Avatar label={initials} />
                            </Skeleton>

                            <CardInfo>
                                <Typography scale={16} weight={500}>
                                    <Skeleton loading={loading}>
                                        {title}
                                    </Skeleton>
                                </Typography>
                                <Typography scale={14}>
                                    <Skeleton loading={loading}>
                                        {subtitle}
                                    </Skeleton>
                                </Typography>
                            </CardInfo>
                            {!loading && (
                                <Button
                                    variant="minimal"
                                    circular
                                    size="sm"
                                    outline={false}
                                    prefix={<Close />}
                                />
                            )}
                        </CardHeader>
                        <CardContent>
                            <Skeleton loading={loading} display="block">
                                <CardImage />
                            </Skeleton>
                        </CardContent>
                        <CardFooter>
                            <Typography scale={14}>
                                <Skeleton loading={loading}>
                                    {description}
                                </Skeleton>
                            </Typography>
                        </CardFooter>
                    </Card>
                </StyledStory>
            </ThemeProvider>
        );
    });
