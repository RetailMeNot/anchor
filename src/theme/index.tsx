// Theme Files
import { typography } from './typography.theme';
import { ColorsTheme } from './colors.theme';
import { ThemeProvider as XstyledThemeProvider } from '@xstyled/styled-components';
import styled from '@xstyled/styled-components';
import { space as spaceStyles, SpaceProps } from '@xstyled/system';
import { BUTTON_KEY, BUTTON_THEME } from '../Button';
import { INPUT_KEY, INPUT_THEME } from '../Form/Input/utils';
import React from 'react';
import {Button} from '../Button';
import classNames from 'classnames';

export const RootTheme = {
    typography,
    radii: { none: '0', base: '4px', modal: '8px', circular: '1000px' },
    [BUTTON_KEY]: BUTTON_THEME,
    [INPUT_KEY]: INPUT_THEME,
    colors: ColorsTheme,
    fonts: {
        base:
            'Avenir Next, Segoe UI, Helvetica Neue, Helvetica, Roboto, sans-serif',
        mono: `'SFMono-Regular', Consolas, Liberation Mono, Menlo, Courier, monospace`,
    },
    // These are all minimums
    breakpoints: {
        xs: 0,
        sm: 576,
        md: 768,
        lg: 992,
        xl: 1200,
        xxl: 1600,
    },
    // todo: we should likely remove the color.borders colors and just use this
    // or find a way to reference the colors.border colors here
    // e.g. `thin solid borders.base`
    borders: {
        light: `thin solid ${ColorsTheme.borders.light}`,
        base: `thin solid ${ColorsTheme.borders.base}`,
        dark: `thin solid ${ColorsTheme.borders.dark}`,
    },

    skeleton: {
        colorStart: '#E7E7E7',
        colorEnd: '#D3D3D3',
    },
};

// Exporting out @xstyled's ThemeProvider. Only way I could get it to work w/o issues.
// export class ThemeProvider2 extends XstyledThemeProvider {}

interface IterableObject {
    [s: number]: number | string | boolean | IterableObject;
  }

interface DebugProps {
    property: number | string | boolean | IterableObject;
    propertyName: string;
    rootProperty?: boolean;
}

interface StyledDebugProps {
    indent?: number;
}

const StyledDebug = styled('div')<StyledDebugProps>`
    font-family: mono;
    margin-top: 0.5rem;
    color: white;

    .category .category h2 {
        background-color: #ec8947;
    }

    .category .category .box {
        border-color: #ec8947;
    }

    .category .category .category h2 {
        background-color: #b84343;
        color: white;
    }

    .category .category .category .box {
        border-color: #b84343;
    }

    .category .category .category .category h2 {
        background-color: #91cdd2;
        color: black;
    }

    .category .category .category .category .box {
        border-color: #91cdd2;
    }

    .category .category .category .category .category h2 {
        background-color: #567224;
        color: white;
    }

    .category .category .category .category .category .box {
        border-color: #567224;
    }
`;

const StyledH2 = styled('h2')`
    ${spaceStyles}
    background-color: #ffea92;
    color: black;
    margin-bottom: 0;
    margin-top: 0;
    padding: 0.25rem 0.5rem 0.25rem 1rem;
    font-size: 1rem;
    cursor: pointer;
    border-radius: 0 1rem 1rem 0;
`;

interface BoxProps extends SpaceProps {
    show?: boolean;
}

const Box = styled('div')<BoxProps>`
    ${spaceStyles}
    width: 20rem;
    border-left: solid #ffea92 2px;
    margin: 0.5rem;
    ${({show}) => `display: ${show ? 'block' : 'none'}`}
`;

const Category = styled('div')<SpaceProps>`
    ${spaceStyles}
    width: 20rem;
`;


const ColorBox = styled('div')<any>`
    width: 1rem;
    height: 1rem;
    display: inline-block;
    vertical-align: middle;
    margin-left: 0.5rem;
    background-color: ${({backgroundColor}) => backgroundColor};
    border: solid white 1px;
`;

const Debug = ({property, propertyName}: DebugProps) => {
    const [show, setShow] = React.useState(false);

    return(
        <StyledDebug>
            {(property || property === 0)
                ? (
                    typeof property === 'number' ||
                    typeof property === 'string' ||
                    typeof property === 'boolean' ||
                    typeof property === 'function'
                        ? (
                                <>
                                    <span style={{fontWeight: 'bold'}}>{propertyName}: </span>
                                    <span>
                                        {typeof property === 'function'
                                            ? 'function'
                                            : /^#[a-zA-Z0-9]{4,7}/.test(property.toString())
                                                ? (<>
                                                    <span style={{fontStyle: 'italic'}}>{property.toString()}</span>
                                                    <ColorBox backgroundColor={property} />
                                                    </>)
                                                : <span style={{fontStyle: 'italic'}}>{property.toString()}</span>
                                        }
                                    </span>
                                </>
                            )
                        :
                            <Category pl={0} className="category">
                                <StyledH2 onClick={() => setShow(!show)}>{propertyName}</StyledH2>
                                <Box pl={2} className="box" show={show}>
                                    {Object.values(property).map((_property, index) => (
                                        <Debug
                                            key={index}
                                            property={_property}
                                            propertyName={Object.getOwnPropertyNames(property)[index]}
                                        />
                                    ))}
                                </Box>
                            </Category>
                    )
                : (
                    <>
                       Empty.
                    </>
                )
            }
        </StyledDebug>
    );
};

const DebugWrapper = styled('div')`
    background-color: rgba(0,0,0,.75);
    border-left-color: transparent;
    position: fixed;
    left: 0;
    top: 0;
    height: 100%;
    overflow: hidden;
    width: 30rem;
    padding: 1rem 2rem 1rem 0;

    &.closed {
        height:2.4rem;
        width: auto;
        padding-right: 1rem;

        .debug-scroll {
            display: none;
        }
    }
`;

const DebugScroll = styled('div')`
    height: 100%;
    overflow-y: auto;
    width: 100%;
`;

interface ThemeProviderProps {
    children?: any;
    debug?: boolean;
    theme?: any;
}

export const ThemeProvider = ({ children, debug, theme }: ThemeProviderProps) => {
    const isOpen = localStorage.themeProviderDebug === 'true';
    const [isOpenState, setIsOpenState] = React.useState(isOpen);

    return (
        <XstyledThemeProvider theme={theme}>
            {debug &&
                <DebugWrapper className={classNames(!isOpenState ? 'closed' : null)}>
                    <Button
                        onClick={() => {
                            localStorage.themeProviderDebug = !isOpenState;
                            setIsOpenState(!isOpenState);
                        }}
                        size="sm"
                    >
                        ThemeProvider Debug
                    </Button>

                    <DebugScroll className="debug-scroll">
                        <Debug property={theme} propertyName="Theme" />
                    </DebugScroll>
                </DebugWrapper>
            }
            {children}
        </XstyledThemeProvider>
    );
};
