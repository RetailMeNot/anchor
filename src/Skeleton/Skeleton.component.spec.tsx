// VENDOR
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { ThemeProvider } from '@xstyled/styled-components';
import { shallow, mount } from 'enzyme';
// ANCHOR
import { RootTheme } from '../theme';
import { Skeleton } from './Skeleton.component';

describe('Component: Skeleton', () => {
    it('should be defined.', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <Skeleton />
            </ThemeProvider>
        );
        const wrapper = mount(subject);
        const component = shallow(subject);

        expect(subject).toBeDefined();
        expect(wrapper).toBeDefined();
        expect(component).toBeDefined();
        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should render in textOnly mode when given only text', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <Skeleton>This is some sample text</Skeleton>
            </ThemeProvider>
        );

        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should wrap over nontext children and hide them', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <Skeleton>
                    <div>hide me!</div>
                </Skeleton>
            </ThemeProvider>
        );

        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should display just the children when not loading', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <Skeleton loading={false}>
                    <div>show me!</div>
                </Skeleton>
            </ThemeProvider>
        );

        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should support multiple color schemes', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <div>
                    <Skeleton colorStart="accent.dark" colorEnd="accent.light">
                        Apple
                    </Skeleton>
                    <Skeleton colorStart="accent.dark" colorEnd="accent.light">
                        Apple
                    </Skeleton>
                    <Skeleton
                        colorStart="secondary.dark"
                        colorEnd="secondary.light"
                    >
                        Banana
                    </Skeleton>
                </div>
            </ThemeProvider>
        );

        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
