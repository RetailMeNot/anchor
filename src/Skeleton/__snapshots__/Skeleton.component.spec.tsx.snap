// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Skeleton should be defined. 1`] = `
.c0 {
  box-sizing: border-box;
  display: inline-block;
  pointer-events: none;
  background: #E7E7E7;
  -webkit-animation: fLEqvv 2s ease-in-out infinite;
  animation: fLEqvv 2s ease-in-out infinite;
}

.c0.c0 * {
  visibility: hidden;
  opacity: 0;
}

<div
  className="c0 anchor-skeleton"
  display="inline-block"
/>
`;

exports[`Component: Skeleton should display just the children when not loading 1`] = `
<div>
  show me!
</div>
`;

exports[`Component: Skeleton should render in textOnly mode when given only text 1`] = `
.c0 {
  box-sizing: border-box;
  display: inherit;
  word-break: break-all;
  color: #E7E7E7;
  -webkit-animation: gXgMws 2s ease-in-out infinite;
  animation: gXgMws 2s ease-in-out infinite;
}

<div
  className="c0 anchor-skeleton"
  display="inherit"
>
  ▆▆▆▆▆▆▆▆▆▆▆▆
</div>
`;

exports[`Component: Skeleton should support multiple color schemes 1`] = `
.c0 {
  box-sizing: border-box;
  display: inherit;
  word-break: break-all;
  color: #0074A6;
  -webkit-animation: gVOVAL 2s ease-in-out infinite;
  animation: gVOVAL 2s ease-in-out infinite;
}

.c1 {
  box-sizing: border-box;
  display: inherit;
  word-break: break-all;
  color: #268068;
  -webkit-animation: gsfEtd 2s ease-in-out infinite;
  animation: gsfEtd 2s ease-in-out infinite;
}

<div>
  <div
    className="c0 anchor-skeleton"
    display="inherit"
  >
    ▆▆▆
  </div>
  <div
    className="c0 anchor-skeleton"
    display="inherit"
  >
    ▆▆▆
  </div>
  <div
    className="c1 anchor-skeleton"
    display="inherit"
  >
    ▆▆▆
  </div>
</div>
`;

exports[`Component: Skeleton should wrap over nontext children and hide them 1`] = `
.c0 {
  box-sizing: border-box;
  display: inline-block;
  pointer-events: none;
  background: #E7E7E7;
  -webkit-animation: fLEqvv 2s ease-in-out infinite;
  animation: fLEqvv 2s ease-in-out infinite;
}

.c0.c0 * {
  visibility: hidden;
  opacity: 0;
}

<div
  className="c0 anchor-skeleton"
  display="inline-block"
>
  <div>
    hide me!
  </div>
</div>
`;
